skops/hub_utils/tests/test_hf_hub.py

```diff
@@ -658,3 +658,51 @@ def test_adding_existing_file_raises(self, init_path, some_file_0):
         )
         with pytest.raises(FileExistsError, match=msg):
             add_files(some_file_0, dst=init_path)
+
+
+class TestUseIntelex:
+    # Tests related to the usage of scikit-learn intelex, see #251
+    def make_config(self, model, requirements, **kwargs):
+        dir_path = tempfile.mkdtemp()
+        shutil.rmtree(dir_path)
+
+        init(
+            model=model,
+            dst=dir_path,
+            task="tabular-classification",
+            data=iris.data,
+            requirements=requirements,
+            **kwargs,
+        )
+        config = get_config(dir_path)
+        return config
+
+    def test_no_intelex(self, classifier):
+        # by default, intelex is not being used
+        config = self.make_config(model=classifier, requirements=["foobar"])
+        environement = config["sklearn"]["environment"]
+
+        assert config["sklearn"]["use_intelex"] is False
+        assert not any(r.startswith("scikit-learn-intelex") for r in environement)
```

Comment:
then we should probably check for the case where the user says `use_intelex=False` but explicitly passes it as a requirement and raise as inconsistent

Reply:
Is that really a good idea? I think if the user wants intelex installed without using it _for the purpose here_, we should allow it.

Reply:
fair point, I guess we don't care if it's installed.